sklearn/feature_extraction/image.py

```diff
@@ -338,6 +338,18 @@ def _extract_patches(arr, patch_shape=8, extraction_step=1):
     return patches
 
 
+@validate_params(
+    {
+        "image": [np.ndarray],
+        "patch_size": [tuple],
+        "max_patches": [
+            Interval(Real, left=0, right=1, closed="neither"),
+            Interval(Integral, left=1, right=None, closed="left"),
```

Comment:
Right now we allow for max_patches=0, in contrary to this code, although the behaviour is buggy see https://github.com/scikit-learn/scikit-learn/issues/25161.

Maybe we can consider this a bug fix and avoid having to do a deprecation cycle?